### PR TITLE
Don't Use Mutable Data Structures Inside Ref

### PR DESCRIPTION
**Pull request by adamgfraser**
We were using a mutable `ArrayBuffer` inside a `Ref`. It wasn't actually causing an issue because the `Ref` wasn't being concurrently updated but that also means we don't really need a `Ref` and I think better not to do unsafe things in our code base unless we have to.

### CODE REVIEW

File: core/shared/src/main/scala/zio/random/package.scala [hunk 1]
```diff
@@ -149,18 +149,17 @@ package object random {
       collection: Collection[A]
     )(implicit bf: BuildFrom[Collection[A], A, Collection[A]]): UIO[Collection[A]] =
       for {
-        bufferRef <- Ref.make(new scala.collection.mutable.ArrayBuffer[A])
-        _         <- bufferRef.update(_ ++= collection)
+        buffer <- ZIO.effectTotal(new scala.collection.mutable.ArrayBuffer[A])
+        _      <- ZIO.effectTotal(buffer ++= collection)
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/random/package.scala:153]: I would merge these two.

File: core/shared/src/main/scala/zio/random/package.scala [hunk 2]
```diff
@@ -149,18 +149,17 @@ package object random {
...
         swap = (i1: Int, i2: Int) =>
-                 bufferRef.update { case buffer =>
+                 ZIO.effectTotal {
                    val tmp = buffer(i1)
                    buffer(i1) = buffer(i2)
                    buffer(i2) = tmp
                    buffer
                  }
         _ <-
           ZIO.foreach((collection.size to 2 by -1).toList)((n: Int) => nextIntBounded(n).flatMap(k => swap(n - 1, k)))
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/random/package.scala:162]: Maybe foreach_?

**Review by mijicd — DISMISSED** [review]
With minor remarks.

File: core/shared/src/main/scala/zio/random/package.scala (hunk 1 above)
Comment (adamgfraser) [reply core/shared/src/main/scala/zio/random/package.scala]: Yes.

File: core/shared/src/main/scala/zio/random/package.scala (hunk 2 above)
Comment (adamgfraser) [reply core/shared/src/main/scala/zio/random/package.scala]: Good idea.

**Comment by adamgfraser** [comment]
@mijicd Incorporated your comments.

**Review by mijicd — DISMISSED** [review]

**Review by mijicd — APPROVED** [review]

**Pull request merged**